src/V3LinkDot.cpp

```diff
@@ -3295,110 +3377,50 @@ class LinkDotResolveVisitor final : public VNVisitor {
                     cextp->v3error("Multiple inheritance illegal on non-interface classes"
                                    " (IEEE 1800-2017 8.13)");
                 }
-                if (cextp->childDTypep() || cextp->dtypep()) {
+                iterate(cextp);
+                if (m_statep->forPrimary()) {
+                    if (cextp->parameterized()) {
+                        // Parameters in extends statement.
+                        // The class can't be resolved in the current pass.
```

Comment:
Should we always resolve all extends in another pass?  Seems cleaner and less likely to have bugs where something breaks when parameters get added.

Reply:
I agree. It would be better to resolve all extends in another pass. Unfortunately, some nodes have to be linked in the first pass to be correctly handled in other V3Link* stages and in V3Param. These are the nodes that are used in the evaluation of parameter values.

I wonder if it's legal to use a member of parameterized base class in expression that initializes a parameter. If it's possible in some specific cases, the current approach is insufficient. If not, I think it should work.